### DISCUSSION TITLE
Why not use more than 4 parts?

### DISCUSSION THREAD

**Discussion by nova-nowiz** [8VIM/8VIM#151] [Ideas]
I'm very mew to 8vim and I was wondering why only 4 parts are used.
Using more would avoid doing a full circle gesture and more letters would be rank 1!
What about letting the user choose the number of parts that he would like to use?

**Comment by Glitchy-Tozier** [comment]
I was wondering the same thing actually.

One obvious answer is that when using like ... 8 sectors, the exit-points in-between the sector lines get smaller and smaller, which might get annoying at high speeds because you might need to slow down to actually leave/enter in the right sector.
Thus, this might slow down the user, instead of speeding him up.

Of course, the problem isn't that big a deal when using something like 5 sectors instead of 4. In that case, i don't see that big of a problem with 5 sectors.

**Comment by flide** [comment]
This was something that I planned to do for Hindi Initially, but have never gotten around to it.
for english we use the 4 parts because it is able to incorporate all letters and then a few symbols as well. That is what the requirement for English is.

**Comment by Glitchy-Tozier** [comment]
In case anyone was wondering: It should be easy to create layouts for layouts with more/less sectors.

I intentionally wrote my little script in a way that lets it calculate layouts for keypads with a different number of sectors.

**Reply by doak** [reply]
> I intentionally wrote my little script in a way that lets it calculate layouts for keypads with a different number of sectors.

Is it available anywhere?

**Comment by doak** [comment]
I could image that the six-armed-xpad (like proposed by @jogrefer in as [_1st Option_ here](https://github.com/flide/8VIM/issues/68#issuecomment-775749834)) works out very well.

**Comment by kjoetom** [comment]
Perhaps a 5-sector layout would be better, firstly in terms of space and secondly maybe also in terms of good flows.

![5-sectors](https://user-images.githubusercontent.com/77550697/155349055-37796b4f-1792-4794-a7a9-5af1dd8a0008.png)
I haven't thought this through though, it's just a thought.

**Reply by lxne** [reply]
I really think that the symmetry of the four sectors and its quadrants are much more suitable for the human brain to comprehend and work with. But off course that is only a sophisticated guess :)

Just imagining to work with 5 or more sectors is giving me some headaches. But that may be because of me being used to the existing 4 sectors.

One thing is obvious. An 8Vim with more sectors will definitely be harder to beginn with. It might keep new users away from an awesome keyboard.

**Reply by kjoetom** [reply]
But maybe it's even easier for our brains to use 5 sectors when we think of this little man with his head, two hands and two feet...

![men](https://github.com/8VIM/8VIM/assets/77550697/68496251-6518-4da1-b4d3-815ae6620936)

In any case, an additional sector would allow more direct and thus shorter paths when writing.

**Comment by foolsbazar** [comment]
Just to throw some opinions against the idea.
I've used a 6 sectors 8pen on another os and I'm not sure that rewriting the core functioning mechanics would really be worth the effort. You can see it in action [here] (https://www.youtube.com/watch?v=ABPYkV7qKq0) and this version in particular was designed so to that you could have any number of sectors, even with non identical sector widths. 
But in the end i found the one with 4 sectors easier to use. 
The problem when increasing the number of sectors is that the length of the entrance to the sector (from the circle) is reduced correspondingly and, with fatty fingers or small screens, it becomes source of mis-strokes. The obvious solution is to increase the size of the circle, but then the space for the swipes is reduced, as well as the swipes themselves become much wider and less confortable. 
![image](https://github.com/8VIM/8VIM/assets/45941292/177f7571-8fe6-4f46-ab33-2577b9bd6291)
That's why, in the end, I was always using the 4 sectors version.

From the little I understood of 8Vim code, i think doing a n-sector based 8pen would require rewriting completely the core mechanics, since I believe it to be based purely on the four cardinals up-down-left-right and it's hard to generalize from that to n-.
